docs/contracts/functions.rst

```diff
@@ -250,33 +249,60 @@ will consume more gas than the 2300 gas stipend:
 - Calling an external function which consumes a large amount of gas
 - Sending Ether
 
-Like any function, the fallback function can execute complex operations as long as there is enough gas passed on to it.
-
-.. warning::
-    The fallback function is also executed if the caller meant to call
-    a function that is not available. If you want to implement the fallback
-    function only to receive ether, you should add a check
-    like ``require(msg.data.length == 0)`` to prevent invalid calls.
-
 .. warning::
     Contracts that receive Ether directly (without a function call, i.e. using ``send`` or ``transfer``)
-    but do not define a fallback function
+    but do not define a receive ether function or a payable fallback function
     throw an exception, sending back the Ether (this was different
     before Solidity v0.4.0). So if you want your contract to receive Ether,
-    you have to implement a payable fallback function.
+    you have to implement a receive ether function (using payable fallback functions for receiving Ether is not recommended).
```

Comment:
```suggestion
    you have to implement a receive ether function (using payable fallback functions for receiving Ether is not recommended because it would not fail on interface confusions).
```